gammapy/estimators/map/tests/test_ts.py

```diff
@@ -277,6 +277,50 @@ def test_compute_ts_map_downsampled(input_dataset):
     assert np.isnan(result["ts"].data[0, 30, 40])
 
 
+def test_ts_map_stat_scan(fake_dataset):
+    model = fake_dataset.models["source"]
+
+    dataset = fake_dataset.downsample(25)
+
+    estimator_ref = TSMapEstimator(
+        model,
+        kernel_width="0.3 deg",
+        energy_edges=[200, 3500] * u.GeV,
+    )
+
+    estimator = TSMapEstimator(
+        model,
+        kernel_width="0.3 deg",
+        selection_optional=["stat_scan"],
+        energy_edges=[200, 3500] * u.GeV,
+    )
+
+    maps_ref = estimator_ref.run(dataset)
+    maps = estimator.run(dataset)
+    success = maps.success.data
+
+    assert maps.stat_scan.geom.data_shape == (1, 109, 2, 2)
+    ts = np.abs(maps["stat_scan"].data.min(axis=1))
+    assert_allclose(ts[success], maps_ref.ts.data[success], rtol=1e-3)
+
+    ind_best = maps.stat_scan.data.argmin(axis=1)
+    ij, ik, il = np.indices(ind_best.shape)
+
+    dnde_ref = maps.dnde_ref.squeeze()
+    assert maps.dnde_scan_values.unit == dnde_ref.unit
+    norm = maps.dnde_scan_values.data[ij, ind_best, ik, il] / dnde_ref.value
+    assert_allclose(norm[success], maps_ref.norm.data[success], rtol=1e-5)
+
+    assert maps.stat_scan.geom.data_shape == (1, 109, 2, 2)
+    ts = np.abs(maps["stat_scan"].data.min(axis=1))
+    assert_allclose(ts[success], maps_ref.ts.data[success], rtol=1e-3)
+
+    ind_best = maps.stat_scan.data.argmin(axis=1)
+    ij, ik, il = np.indices(ind_best.shape)
+    dnde = maps.dnde_scan_values.quantity[ij, ind_best, ik, il]
+    assert_allclose(dnde[success], maps_ref.norm.data[success] * dnde_ref, rtol=1e-5)
```

Comment:
Technically this is the same test as l311-312, right?

Reply:
yes removed